api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -73,6 +73,26 @@ const (
 	TopologyManagerPolicySingleNumaNode TopologyManagerPolicy = "single-numa-node"
 )
 
+// TransparentHugePage enumerates the values for various modes of Transparent Hugepages.
+type TransparentHugePage string
+
+const (
```

Comment:
Ah I just noticed that these are options for both `transparentHugePageEnabled` and `transparentHugePageDefrag`. It may be due to my own confusion, but would a name like `TransparentHugePageOptions` be more clear? This is just a suggestion though and I think `TransparentHugePage` works as is.

Reply:
`TransparentHugePageOptions` makes more sense from the readability perspective. Making the changes...

Reply:
Updated `TransparentHugePage` to TransparentHugePageOption`